cmd/internal/cli/build_linux.go

```diff
@@ -39,6 +39,10 @@ import (
 
 func fakerootExec(isDeffile bool) {
 	useSuid := starter.IsSuidInstall()
+	if os.Getenv("_APPTAINER_E2E_NOT_USE_SUID") != "" {
```

Comment:
Rather than having custom environment variables for testing, I asked Cedric and he suggested adding hidden command line option(s) in cmd/internal/build.go, with associated environment variables.  That gives the most consistency in dealing with environment variables, and code in cmd/internal/cli can just refer to the flag's variable name instead of reading an environment variable at all.

Cedric suggested a single hidden option called `--fakeroot-mode` but I'm thinking more generically.  I suggest more than one option:
1. A non-hidden `--userns` option, to match what is used with action flags, except there can't be a shorthand `-u` because it is already taken.  Then `APPTAINER_USERNS=1` can be set for modes 1b, 2, and 3.
2. A hidden `--ignore-subuid` option to ignore `/etc/subuid` entries.  Then `APPTAINER_IGNORE_SUBUID=1` can be set for modes 2 through 4.
3. A hidden `--ignore-fakeroot-command` option to ignore the fakeroot command.  Then `APPTAINER_IGNORE_FAKEROOT_COMMAND=1` can be set for mode 3.
4. A hidden `--ignore-userns` option to ignore user namespaces.  Then `APPTAINER_IGNORE_USERNS` can be set for mode 4.

Since the `--fakeroot` option is also supported in the same way for action commands, I think it would be good at the same time to add the hidden options 2 through 4 above also to the action flags.

Reply:
@DrDaveD the code is refactored. Thanks for the comments!